src/sentry/debug_files/artifact_bundles.py

```diff
@@ -169,6 +172,35 @@ def _index_urls_in_bundle(
 # ===== Renewal of Artifact Bundles =====
 
 
+@sentry_sdk.tracing.trace
+def maybe_renew_artifact_bundles_from_processing(project_id: int, used_artifact_bundles: List[str]):
+    if options.get("symbolicator.sourcemaps-bundle-index-refresh-sample-rate") <= random.random():
+        return
+
+    artifact_bundle_ids = []
+    for download_id in used_artifact_bundles:
+        # the `download_id` is in a `artifact_bundle/$ID` format
+        split = download_id.split("/")
+        if len(split) < 2:
+            continue
+        ty, ty_id, *_rest = split
+        if ty != "artifact_bundle":
+            continue
+        artifact_bundle_ids.append(ty_id)
+
+    # FIXME: This function is being called for every processed event, so ideally
+    # we would heavily debounce this and avoid doing such a query directly.
```

Comment:
Is this an obstacle to merging this or just an opportunity for optimization?

Reply:
I gated the function behind an option, so we can crank that up and down depending on the load.
It is hard to get a feeling for how expensive this would be, so we will have to try and see.

Reply:
Yeah, makes sense.